explorer/interpreter/interpreter.cpp

```diff
@@ -1075,6 +1083,46 @@ auto Interpreter::CallFunction(const CallExpression& call,
   }
 }
 
+// -
+auto Interpreter::StepInstantiateType() -> ErrorOr<Success> {
+  Action& act = todo_.CurrentAction();
+  Nonnull<const Value*> type = cast<TypeInstantiationAction>(act).type();
+  SourceLocation source_loc = cast<TypeInstantiationAction>(act).source_loc();
+
+  switch (type->kind()) {
+    case Value::Kind::NominalClassType: {
```

Comment:
Is it worth redefining those, in addition to the ones in `InstantiateType`? I'm slightly worried we might be duplicating code between both

Reply:
Oh I didn't notice that, I think we should remove those two from `InstantiateType` method as both `NominalClassType` and `PointerType` need to instantiate the type again and it should spawn an action but in `InstantiateType` method it directly evaluates by calling `InstantiateType` directly.

What do you suggest?

Reply:
That's probably the right approach is we lean towards a `Step` based intantiation, yes!